blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/scm/ScmFile.java

```diff
@@ -0,0 +1,12 @@
+package io.jenkins.blueocean.rest.impl.pipeline.scm;
+
+import org.kohsuke.stapler.export.ExportedBean;
+
+/**
+ * Content stored in SCM
+ *
+ * @author Vivek Pandey
+ */
+@ExportedBean(defaultVisibility = 999999)
+public abstract class ScmFile{
```

Comment:
Prefer abstract class over interface unless it describes behavior, gives leverage in case of binary incompatible change later on.